File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start == 1) {
        return 0;
    } else if (start % 2 == 0) {
        return 1 + steps(start / 2);
    } else {
        return 1 + steps(start * 3 + 1);
    }
}

File: solutions/fsharp/collatz-conjecture/1/CollatzConjecture.fs
module CollatzConjecture

let rec step_aux n l = 
    match n with
    | 1 -> l
    | i when i % 2 = 0 -> step_aux (n / 2) (l + 1)
    | i -> step_aux (3 * n + 1) (l + 1)
let steps (number: int): int option = 
    match number with
    | n when n <= 0 -> None
    | n -> step_aux n 0 |> Some